packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -284,6 +284,12 @@ export default class NewVm extends BaseComponent {
     this._reset()
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps.template !== this.props.template) {
```

Comment:
You should only check if the ID changed because if something else changed, the form will completely re-initialize and override everything the user has changed.